preprocessing.py

```diff
@@ -6,33 +6,36 @@
 import os
 
 
-def one_doc_preprocess(ar): #This is dataset-specific code. I think that you should name this method as "cacm_doc_preprocess"
+def one_doc_preprocess(ar, list_of_arguments): 
     tree = ET.parse(ar)
     root = tree.getroot()
     stop_words = set(stopwords.words('english'))
     translator = str.maketrans('', '', string.punctuation)
     ps = PorterStemmer()
-    a = [elem.tag for elem in root.iter()] #for what reasons you created a-variable and didn't use it?
-    for docid in root.iter("DOCID"): #Do I understand correctly that one XML is containing only one document (one DOCID, title, body) ?
-        name = docid.text.replace(" ", "") #what if document doesn't contain name, title or other fields? I think that you should prevent strange errors with format checking 
-                                           # like if name == None then exit with message "document has invalid format"
+    for docid in root.iter("DOCID"): #Do I understand correctly that one XML is containing only one document (one DOCID, title, body) ? Yes
+        name = docid.text.replace(" ", "")
+        if docid == None:
```

Comment:
hmm.... if doc_id doesn't contains in xml then "docid.text.replace(" ", "")" throws exception earlier than this check. You should check this condition before manipulations with docid